registry/azure.go

```diff
@@ -0,0 +1,53 @@
+package registry
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"strings"
+)
+
+const (
+    // Mount volume from hostpath.
+    azureCloudConfigJsonFile = "/etc/kubernetes/azure.json"
+)
+
+type azureCloudConfig struct {
+    AADClientId string `json:"aadClientId"`
+    AADClientSecret string `json:"aadClientSecret"`
+}
+
+// Fetch Azure Active Directory clientid/secret pair from azure.json, usable for container registry authentication.
+//
+// Note: azure.json is populated by AKS/AKS-Engine script kubernetesconfigs.sh. The file is then passed to kubelet via
+// --azure-container-registry-config=/etc/kubernetes/azure.json, parsed by kubernetes/kubernetes' azure_credentials.go
+// https://github.com/kubernetes/kubernetes/issues/58034 seeks to deprecate this kubelet command-line argument, possibly
+// replacing it with managed identity for the Node VMs. See https://github.com/Azure/acr/blob/master/docs/AAD-OAuth.md
+func GetAzureCloudConfigAADToken(host string) (creds, error) {
```

Comment:
I don't think this or `HostIsAzureContainerRegistry` need to be exported -- they are just used in the tests and in `credsFor`, both in the same package.

Reply:
I was matching the pattern of `GetGCPOauthToken` being exported - it's also only used in those two places in the `registry` package. I wasn't sure if that was intentional. I can make this private anyway.

Reply:
Fair point, following the apparent convention ought not earn you demerits :-)